### DISCUSSION TITLE
New Tools for Your Toolkit: Check Out What’s Fresh with GitHub Copilot!

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#142249] [Copilot News and Announcements]
![unnamed](https://github.com/user-attachments/assets/66bbbeac-b07b-4912-ad07-fc6ad33c2a49)

### 🚀 We’ve Got Some Game-Changing Updates Across the Board! 🎉

From leveling up completions for C# and C++ --  we’re excited to share these latest enhancements for GitHub Copilot. We want to hear your thoughts—so dive in, explore these updates, and let us know what’s making a difference in your workflow! 👇

---

### 🌟 Smarter Completions for C++ in Visual Studio

**Attention C++ developers!** GitHub Copilot now leverages related files (like headers) to provide contextually relevant suggestions—even if they’re not open. This update, available on Visual Studio 2022 (v17.12+), brings sharper, more accurate completions and cuts down on off-target suggestions. 

**Have you noticed a boost in completion accuracy?** Drop your thoughts below, and let’s keep improving the Copilot experience together! 

---

### 🖥️ Terminal Chat in Windows Terminal Canary

CLI power users, this one’s for you! GitHub Copilot is now in Windows Terminal, bringing real-time command suggestions and explanations directly to your terminal window with Terminal Chat. Whether you need help with command syntax or a quick breakdown of a command’s purpose, Copilot’s got you covered without the need to switch screens.

To get started:

Install Windows Terminal Canary.
Check out the [Terminal Chat documentation](https://learn.microsoft.com/windows/terminal/terminal-chat) to get connected.
💼 Copilot Business and Enterprise users: Remember, your org admin will need to enable CLI access in settings.

How has Copilot made your terminal workflows easier? Share your tips, tricks, and feedback below!

---

### ⚙️ Enhanced .NET Completions in Visual Studio for C# Devs

Hey C# devs, imagine Copilot providing even more spot-on suggestions by looking at semantically relevant files that aren’t open in your editor! Now available in Visual Studio 2022 (v17.11+), this update cuts down on those "hallucinations" and delivers more precise completions for a smoother coding flow. For more details, visit the .NET team blog [here](https://devblogs.microsoft.com/dotnet/improving-github-copilot-completions-in-visual-studio-for-csharp-developers/).

**How’s it working for you?** Are the suggestions hitting the mark? Let’s talk about what you’re noticing below!

---

Each of these updates is a step toward making GitHub Copilot work harder and smarter for you. **Have a favorite?** Or ideas for what should come next? Let’s get the conversation going! 💬

**Comment by Kurnianazara10** [comment]
**mantap sekali kopilot**

**Comment by amlel-el-mahrouss** [comment]
👍🏼  It's cool!
But what about the hallucinations on header files?
for example it reports a type which doesn't exist.
such as: `ns::string_builder` or something

**Comment by peachmontt** [comment]
That sounds like a fantastic update for C++ developers using GitHub Copilot in Visual Studio! The ability for Copilot to consider related files, including headers, even when they're not open, can significantly enhance the coding experience by providing more contextually relevant completions. This change is likely to help reduce the number of off-target suggestions, making coding more efficient and accurate.

**Comment by pipe01** [comment]
Does this also work on C files?

**Reply by peachmontt** [reply]
should work
